src/Platforms/AbstractMySQLPlatform.php

```diff
@@ -460,6 +460,11 @@ private function buildTableOptions(array $options): string
 
         // Charset
         if (! isset($options['charset'])) {
+            Deprecation::trigger(
+                'doctrine/dbal',
+                'https://github.com/doctrine/dbal/pull/5278',
+                'Omitting the charset option is deprecated, be explicit about it instead. We recommend using "utf8b4"'
+            );
             $options['charset'] = 'utf8';
```

Comment:
> The charset and collation parameters are optional because they are not portable and are not required in the CREATE TABLE DDL.

@morozov ok, so although this is not the right way to handle the problem, you do admit there is an issue on this line, right? If the charset parameter is optional, why is there a default value here?

Reply:
> you do admit there is an issue on this line, right?

Yes, it will be addressed by https://github.com/doctrine/dbal/pull/4644.

> If the charset parameter is optional, why is there a default value here?

I don't know.